controller/attribute/util.py

```diff
@@ -106,3 +121,47 @@ def run_attribute_calculation_exec_env(
     s3.delete_object(org_id, project_id + "/" + prefixed_payload)
 
     return calculated_attributes
+
+
+def __read_container_logs_thread(
+    project_id: str,
+    container_name: str,
+    attribute_id: str,
+    docker_container: Any,
+) -> None:
+
+    ctx_token = general.get_ctx_token()
+
+    attribute_item = attribute.get(project_id, attribute_id)
+    c = 0
+    while container_name in __containers_running:
+        time.sleep(1)
+        c += 1
+        if c > 100:
+            ctx_token = general.remove_and_refresh_session(ctx_token, True)
+            attribute_item = attribute.get(project_id, attribute_id)
+        if not container_name in __containers_running:
+            break
+        try:
+            last_progress = 0.0
+            for log in docker_container.logs(
+                stream=True,
+                tail=10
```

Comment:
during my testing for lf progress, I noticed that tail doesn't really work well with user logs (so if someone prints something in the log on their own). So e.g., if someone prints more than 10 between the progress logs, we won't be able to update. 
Further the logs seem to be doubled? 
![image](https://github.com/code-kern-ai/refinery-gateway/assets/91723236/a01b193a-1e38-45ab-844f-9bd3f5267f3b)


Why was decided to change this from the original logic with the timestamp?

Reply:
Ofc this can lead to problems when someone prints, I could just extend the size of the tail, but there could still be edge cases in which the progress logs will not be collected, but this is imo no real problem since not every progress step is necessary for a fully functional progress overview. 

the timestamp logic is quite complicated to first understand and then refactor imo, so I thought about another solution and came up with this